### PR TITLE
feat(auth): pass tax amount into PayPal DoReferenceTransaction

### PR DESCRIPTION
**Pull request by biancadanforth**
Because:

* We want to charge users sales tax and ensure it shows up on PayPal invoices.

This commit:

* Passes the optional tax amount argument into DoReferenceTransaction along with other amounts now required by PayPal when tax is added.

Closes #FXA-5765

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Screen Shot 2022-10-20 at 12 35 35 PM

## Other information (Optional)

DRYed up the `processInvoice` describe block while I was at it.

### CODE REVIEW

File: packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts [hunk 1]
```diff
@@ -235,7 +235,7 @@ export class PayPalHandler extends StripeWebhookHandler {
         await this.paypalHelper.processInvoice({
           customer,
           invoice: latestInvoice,
-          ipaddress: request.info.remoteAddress,
+          ipaddress: request.app.clientAddress,
```
Comment (biancadanforth) [inline packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts:238]: This is an unrelated fix I'm sneaking in here -- we should be sending PayPal the client IP address, not the remote address. See `IPADDRESS` in the [`DoReferenceTransaction` docs](https://developer.paypal.com/api/nvp-soap/do-reference-transaction-nvp/).

**Review by bbangert — APPROVED** [review]
LGTM!
